teuthology/task/kernel.py

```diff
@@ -682,7 +682,7 @@ def wait_for_reboot(ctx, need_install, timeout, distro=False):
     starttime = time.time()
     while need_install:
         teuthology.reconnect(ctx, timeout)
-        for client in need_install.keys():
+        for client in list(need_install.keys()):
```

Comment:
Is that change needed? Iterating over keys should just work fine in py3, too.

Reply:
Yes, it is needed very much because of:
```
RuntimeError: dictionary changed size during iteration
```

Reply:
btw, there is a shorter way of getting keys as a list from the dict, I would write like `list(need_install)` but this is probably not so intuitive as with exclusively using keys.